### PR TITLE
GH-37639: [CI] Fix checkout on older OSes

### LINKED ISSUE #37639: [CI] actions/checkout@v4 breaks on older OSes due to node 20
### Describe the bug, including details regarding any error messages, version, and platform.

e.g. https://github.com/ursacomputing/crossbow/actions/runs/6127061836/job/16632216661
https://github.com/ursacomputing/crossbow/actions/runs/6127059010/job/16632383891

We should add an option in the jinja macro to use v3

### Component(s)

Continuous Integration

### PR DESCRIPTION
**Pull request by assignUser**
### Rationale for this change

Jobs are failing due to not being supported by node 20

### What changes are included in this PR?

Using v3 where needed.

### Are these changes tested?

Crossbow
* Closes: #37639

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages homebrew-r-autobrew

File: dev/tasks/macros.jinja [hunk 1]
```diff
@@ -25,9 +25,9 @@ on:
       - "*-github-*"
 {% endmacro %}
 
-{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive") -%}
+{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive", action_v="v4") -%}
   - name: Checkout Arrow
-    uses: actions/checkout@v4
+    uses: actions/checkout{{ action_v }}
```
Comment (raulcd) [inline dev/tasks/macros.jinja:30]: isn't the `@` missing? I would probably also add the v and only pass the `action_version=4`
```suggestion
    uses: actions/checkout@v{{ action_version }}
```

File: dev/tasks/r/github.macos.autobrew.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
             - "{{ macros.r_release.ver }}"
             - "{{ macros.r_oldrel.ver }}"
     steps:
-      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.github_checkout_arrow(action_v='v3')|indent }}
```
Comment (assignUser) [inline dev/tasks/r/github.macos.autobrew.yml:37]:
```suggestion
      {{ macros.github_checkout_arrow(action_v='3')|indent }}
```

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages homebrew-r-autobrew

File: dev/tasks/macros.jinja [hunk 3]
```diff
@@ -25,9 +25,9 @@ on:
       - "*-github-*"
 {% endmacro %}
 
-{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive") -%}
+{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive", action_v="v4") -%}
   - name: Checkout Arrow
-    uses: actions/checkout@v4
+    uses: actions/checkout@v{{ action_version }}
```
Comment (assignUser) [inline dev/tasks/macros.jinja:30]:
```suggestion
    uses: actions/checkout@v{{ action_v }}
```

File: dev/tasks/macros.jinja [hunk 4]
```diff
@@ -25,9 +25,9 @@ on:
       - "*-github-*"
 {% endmacro %}
 
-{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive") -%}
+{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive", action_v="v4") -%}
```
Comment (assignUser) [inline dev/tasks/macros.jinja:28]:
```suggestion
{%- macro github_checkout_arrow(fetch_depth=1, submodules="recursive", action_v="4") -%}
```

**Comment by assignUser** [comment]
Failures are unrelated (#37610 )

**Comment by kou** [comment]
Could you rebase on main?

**Comment by assignUser** [comment]
@github-actions crossbow submit r-binary-packages homebrew-r-autobrew

**Comment by assignUser** [comment]
New, exciting erros :grin:  but still unrelated

**Review by raulcd — APPROVED** [review]
The change looks good to me. Are we tracking the existing CI errors on an opened issue?

**Comment by assignUser** [comment]
I created one: https://github.com/apache/arrow/issues/37685 and the other one seems to be this issue previously happening on windows? https://github.com/apache/arrow/issues/29563

**Pull request merged**